oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/worker/RegisterPersistentWorker.java

```diff
@@ -80,43 +92,80 @@ private void onWork(RegisterSource registerSource) {
         } else {
             sources.get(registerSource).combine(registerSource);
         }
-
         if (sources.size() > 1000 || registerSource.getEndOfBatchContext().isEndOfBatch()) {
             sources.values().forEach(source -> {
+                RegisterSource cacheSource = null;
                 try {
-                    RegisterSource dbSource = registerDAO.get(modelName, source.id());
+                    RegisterSource dbSource = this.getRegisterSourceCache(source.id());
                     if (Objects.nonNull(dbSource)) {
                         if (dbSource.combine(source)) {
-                            registerDAO.forceUpdate(modelName, dbSource);
```

Comment:
forceUpdate means update immediately. I am not sure that it can be change to batch. Because the data will be delay to visible to others.

Reply:
I've used the IMMEDIATE RefreshPolicy in my code, and you can find it in ElasticSearchClient.

Reply:
I means cache could cause dirty read in cluster mode. As @wu-sheng say.